Source/WebCore/platform/text/cocoa/LocalizedDateCache.mm

```diff
@@ -155,10 +155,15 @@ static void _localeChanged(CFNotificationCenterRef, void*, CFStringRef, const vo
     ASSERT(numberOfGregorianMonths == 12);
 
     // For each month (in the Gregorian Calendar), format a date and measure its length.
+    // FIXME: Week is checked here as a workaround for an issue which caused the displayed text to overflow onto a second line.
```

Comment:
I don't think it's a workaround? From my understanding, it's essential and something we intend to keep around.

Reply:
Isn't that code meant to calculate the maximum width based on month names, though? I thought it was a workaround since the month names aren't displayed on the week picker's text (i.e "Week 29 2024")

Reply:
It's not specifically about whether the month is displayed or not, it's just whether the month can influence the width of the displayed string which is true for weeks.

It could probably be further "optimized", since we know the maximum width would be for months with two-digit weeks. But this is not particularly hot code and I don't think it's something we need to come back to once this patch has landed.